server/src/test/java/org/apache/druid/metadata/SqlSegmentsMetadataManagerTest.java

```diff
@@ -283,7 +283,9 @@ public void testPollPeriodicallyAndOnDemandInterleave() throws Exception
     Assert.assertTrue(sqlSegmentsMetadataManager.getLatestDatabasePoll() instanceof SqlSegmentsMetadataManager.PeriodicDatabasePoll);
     dataSourcesSnapshot = sqlSegmentsMetadataManager.getDataSourcesSnapshot();
     Assert.assertEquals(
-        ImmutableList.of("wikipedia3", "wikipedia", "wikipedia2"),
+    // This test is guava version sensitive when upgrading to guava > 32
+    // the order changes to
```

Comment:
it sounds like it would be less brittle to change this test to compare sets instead of lists. I don't think anything relies on ordering here.

Reply:
This came up in:
https://github.com/apache/druid/pull/14767 
There is an effort on de-flaking tests, I would guess this will show up on their radar sooner or later.

Reply:
given that we changed the expected order once already I suggest we move to a set. I made suggestions you should be able to merge (mostly) as is.

Reply:
Thank you!